Add Alternative Authentication Method. (#859)
FusionAuth/fusionauth-site: site/docs/v1/tech/oauth/endpoints.adoc

> Do not send these parameters in a query string or as JSON.

Reviewer: We should also probably modify the "You must provide client authentication using the Basic Authentication Scheme authorization header." verbiage in the request header section too.

[.api]
[field]#client_id# [type]#[String]# [optional]#Optional#::
The unique client identifier.
The client Id is the Id of the FusionAuth Entity in which you are attempting to authenticate.
+
This parameter is optional when the `Authorization` header is provided.

> When the `Authorization` header is not provided, this parameter is then required.

Reviewer: ```suggestion
When the `Authorization` header is not provided, this parameter is required.
```

> [field]#client_secret# [type]#[String]# [optional]#Optional# [since]#Available since 1.5.0#::

Reviewer 2: The client credentials is new, so we don't need a version flag on this one.

The client secret of the Entity.

> This value may optionally be provided in the request body instead of the `Authorization` header documented above.

Reviewer: ```suggestion
This parameter is optional when the `Authorization` header is provided.
When the `Authorization` header is not provided, this parameter is required.
```

Would re-use the same verbiage as used with `client_id`.

include::docs/v1/tech/oauth/_client_credentials_example_json_response.json[]
----

> 

Reviewer: Nit, but not sure what this extra space buys us :)
